### PR TITLE
doc: nrf_modem: small updates

### PR DESCRIPTION
**Pull request by lemrey**
Small updates to the documentation.

Signed-off-by: Emanuele Di Santo <emdi@nordicsemi.no>

### CODE REVIEW

File: nrf_modem/doc/fault_handling.rst [hunk 1]
```diff
@@ -28,6 +28,13 @@ The application can reinitialize the modem by reinitializing the Modem library t
 
    The reinitialization of the Modem library must be done outside of the Modem library fault handler.
 
+.. important::
+   If modem traces are enabled, the modem will output keep outputting trace data (a coredump) upon a fault, after having signalled the fault to the application.
```
Comment (rlubos) [inline nrf_modem/doc/fault_handling.rst:32]: `the modem will output keep outputting` - this doesn't sound good.

Comment (eivindj-nordic) [inline nrf_modem/doc/fault_handling.rst:32]: Maybe
```suggestion
   If modem traces are enabled, the modem will perform a coredump upon a modem fault. The trace data is output by the modem after signaling the fault to the application.
```

Comment (divipillai) [inline nrf_modem/doc/fault_handling.rst:32]:
```suggestion
   If modem traces are enabled, the modem continues to output trace data (a coredump) in the event of a fault, even after signaling the application of the fault.
```

File: nrf_modem/doc/fault_handling.rst [hunk 2]
```diff
@@ -28,6 +28,13 @@ The application can reinitialize the modem by reinitializing the Modem library t
...
+   To make sure the application is able to retrieve the coredump correctly, the application must not re-initialize the modem until all the outstanding trace data has been processed.
+   The :c:func:`nrf_modem_trace_get()` function will return ``-ENODATA`` when all outstanding trace data has been processed by the application.
+
+When using the Modem library in |NCS|, the Modem library integration layer takes care of synchronizing the modem re-initialization with the tracing operations.
```
Comment (divipillai) [inline nrf_modem/doc/fault_handling.rst:36]:
```suggestion
When the Modem library is used in |NCS|, the :ref:`nrf_modem_lib_readme` handles synchronizing modem re-initialization with tracing operations.
```

**Review by eivindj-nordic — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**